File: tests/fixtures/ruby-pilot/host/test/invoice_service_test.rb
# frozen_string_literal: true

require_relative "../lib/billing/invoice_service"

service = Billing::InvoiceService.new
actual = service.render("INV-42", 50_000)
expected = "invoice:INV-42:125"
raise "expected #{expected.inspect}, got #{actual.inspect}" unless actual == expected

puts "native-test:ok"

File: tests/fixtures/dart-pilot/host/test/invoice_service_test.dart
import '../lib/invoice_service.dart';

void main() {
  const service = InvoiceService();
  final actual = service.render((id: 'INV-42', cents: 125));
  if (actual != 'invoice:INV-42:125') {
    throw StateError('unexpected invoice rendering: $actual');
  }
  print('native-test:ok');
}
